src/main/java/org/opendevstack/provision/services/ConfluenceAdapter.java

```diff
@@ -161,17 +171,36 @@ protected String getJiraServerId() throws IOException {
     return jiraServerId;
   }
 
-  protected String getBluePrintId() throws IOException {
+  protected String getBluePrintId(String projectTypeKey) throws IOException 
+  {
     String bluePrintId = null;
     String url = String.format(BLUEPRINT_PATTERN, confluenceUri, confluenceApiPath);
     List<Object> blueprints =
         getList(url, crowdCookieValue, new TypeReference<List<Blueprint>>() {});
+    
+    String confluencetemplateKeyPrefix = "confluence.blueprint.key.";
+
+    String template =
+    	(projectTypeKey != null && !projectTypeKey.equals(defaultProjectKey) && 
+    		environment.containsProperty(confluencetemplateKeyPrefix + projectTypeKey) && 
+    		projectTemplateKeyNames.contains(projectTypeKey)) ?
+    		environment.getProperty(confluencetemplateKeyPrefix + projectTypeKey) : 
+    			confluenceBlueprintKey;
```

Comment:
Maybe we can extract the condition, or even returning the whole template string into a private method, I find this hard to read ...

Reply:
For this kind of logic, I sometimes add a public static method because it makes the its logic easy to unit test... it would look like: 
public static String resolveTemplate(String projectTypeKey, String defaultProjectKey, Environment environment
however in this case because the refactored method would have more than 5 params, I would consider to introduce a nested public static helper class maybe called TemplateResolver() and pass environment or other dependencies in the constructor.

Reply:
@stitakis  /  @michaelsauter  - I can do what I did in the jiraadapter - to extract it and comment it -good enough?

Reply:
yep, sounds good to me :-)